src/context/src/rendering_context.rs

```diff
@@ -0,0 +1,90 @@
+extern crate amethyst_renderer;
+extern crate glutin;
+extern crate gfx_window_glutin;
+extern crate gfx_device_gl;
+extern crate gfx;
+
+use amethyst_config::Element;
+use std::path::Path;
+use self::glutin::{Window, WindowBuilder};
+use self::amethyst_renderer::Renderer;
+use self::amethyst_renderer::target::{ColorFormat, DepthFormat};
+use self::gfx_device_gl::{Device, Factory, Resources, CommandBuffer};
+use self::gfx::handle::{RenderTargetView, DepthStencilView};
+
+config!(
+    /// Contains display config,
+    /// it is required to create a `(Window, RenderingContext)` pair
+    struct DisplayConfig {
+        pub title: String = "Amethyst game".to_string(),
+        pub fullscreen: bool = false,
+        pub dimensions: Option<(u32, u32)> = None,
+        pub min_dimensions: Option<(u32, u32)> = None,
+        pub max_dimensions: Option<(u32, u32)> = None,
+        pub vsync: bool = true,
+        pub multisampling: u16 = 1,
+        pub visibility: bool = true,
+    }
+);
+
+/// Contains resources required for rendering
+pub struct RenderingContext {
+    pub device: Device,
+    pub factory: Factory,
+    pub renderer: Renderer<Resources, CommandBuffer>,
+    pub main_color: RenderTargetView<Resources, ColorFormat>,
+    pub main_depth: DepthStencilView<Resources, DepthFormat>,
+}
+
+impl RenderingContext {
+    //! Creates a `(Window, RenderingContext)` pair configured according to `DisplayConfig`
+    pub fn new(display_config: DisplayConfig) -> (Window, RenderingContext) {
+        let title = display_config.title;
+        let multisampling = display_config.multisampling;
+        let visibility = display_config.visibility;
+
+        let mut builder = WindowBuilder::new()
+            .with_title(title)
+            .with_multisampling(multisampling)
+            .with_visibility(visibility);
+
+        match display_config.dimensions {
```

Comment:
Since the None never actually does something, I think it might be better to just make this a 
```rust
if let Some((w, h)) = display_config.dimensions {
    builder = builder.with_dimensions(w, h);
}
```